### ISSUE TITLE
`valid-expect` should handle `expect.soft` as a Vitest expect API

### ISSUE THREAD

**Issue by nami8824** [vitest-dev/eslint-plugin-vitest#921]
## Summary

`valid-expect` currently appears to treat `expect.soft(...)` as if `soft` were the matcher.

However, in Vitest, `expect.soft` is an expect API that works similarly to `expect`, except failed assertions are recorded and the test continues running.

```ts
expect.soft(value).toBe(1)
```

In this chain, `soft` is not the matcher. The actual matcher is `toBe`.

This can cause `valid-expect` to miss invalid `expect.soft` usages.

## Reproduction

Add the following case to the `invalid` section of `tests/valid-expect.test.ts`:

```ts
{
  code: 'expect.soft("something");',
  errors: [{ endColumn: 25, column: 1, messageId: 'matcherNotFound' }],
},
```

Then run:

```sh
pnpm test tests/valid-expect.test.ts
```

Currently, this case fails because `valid-expect` reports no errors:

```txt
AssertionError: Should have 1 error but had 0: []

0 !== 1
```

For comparison, the equivalent plain `expect` case is already reported correctly:

```ts
{
  code: 'expect("something");',
  errors: [{ endColumn: 20, column: 1, messageId: 'matcherNotFound' }],
},
```

## Expected behavior

`valid-expect` should treat `expect.soft(...)` like `expect(...)` for matcher validation.

For example:

```ts
expect.soft("something")
```

should be reported as `matcherNotFound`.

And this should remain valid:

```ts
expect.soft("something").toBe("something")
```
